.github/workflows/gh-pages.yml

```diff
@@ -13,29 +13,29 @@ jobs:
         with:
           submodules: true
 
-          # Does not provide asciidoctor out of the box :(
       - name: Setup Hugo
         uses: peaceiris/actions-hugo@v2
         with:
           hugo-version: 'latest'
 
-      - name: Setup asciidoctor
-        run: |
-          sudo gem update --system 3.0.6 # cf. https://github.com/rubygems/rubygems/issues/3068#issuecomment-574775885
-          sudo gem install asciidoctor
-          sudo .github/scripts/reroute_asciidoctor.sh
-          ruby --version; gem --version; asciidoctor --version
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+
+      - run: bundle config set path 'vendor/bundle'
+      - run: bundle install
+      - run: bash .github/scripts/reroute_asciidoctor.sh
```

Comment:
BTW, I think we can just replace the script with the following alias.

```diff
- sudo .github/scripts/reroute_asciidoctor.sh
+ alias asciidoctor="asciidoctor --attribute=experimental=true --attribute=icons=font"
```

Or run it before running Hugo.